conda_smithy/templates/run_docker_build.sh.tmpl

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+# PLEASE NOTE: This script has been automatically generated by conda-smithy. Any changes here
+# will be lost next time ``conda smithy rerender`` is run. If you would like to make permanent
+# changes to this script, consider a proposal to conda-smithy so that other feedstocks can also
+# benefit from the improvement.
+
+set -xeuo pipefail
+
+FEEDSTOCK_ROOT=$(cd "$(dirname "$0")/.."; pwd;)
+RECIPE_ROOT=$FEEDSTOCK_ROOT/{{ recipe_dir }}
+
+docker info
+
+# In order for the conda-build process in the container to write to the mounted
+# volumes, we need to run with the same id as the host machine, which is
+# normally the owner of the mounted volumes, or at least has write permission
+export HOST_USER_ID=$(id -u)
+# Check if docker-machine is being used (normally on OSX) and get the uid from
+# the VM
+if hash docker-machine 2> /dev/null && docker-machine active > /dev/null; then
+    export HOST_USER_ID=$(docker-machine ssh $(docker-machine active) id -u)
+fi
+
+ARTIFACTS="$FEEDSTOCK_ROOT/build_artifacts"
+
+if [ -z "$CONFIG" ]; then
+    echo "Need to set CONFIG env variable"
+    exit 1
+fi
+
+test -d "$ARTIFACTS" || mkdir "$ARTIFACTS"
```

Comment:
Was there a reason we want to do this instead of `mkdir -p "$ARTIFACTS"`?

cc @pkgw